### DISCUSSION TITLE
CONTRIBUTORS.md

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#798] [Ideas]
What do you think about a public CONTRIBUTORS.md file? Something like:

```markdown
# Contributors to PyPDF2

The following people made PyPDF2 in its current form possible - by writing code, documentation, tests, reporting issues, drafting new features, supporting the maintainers with answering questions:

* Last name, First name: 140-characters of text; links to linkedin / github / other profiles and personal pages are ok
* Last name, First name: 140-characters of text; links to linkedin / github / other profiles and personal pages are ok
* Last name, First name: 140-characters of text; links to linkedin / github / other profiles and personal pages are ok
```

The goal is to encourage more people to help in various ways.

edit: Examples are:

* https://github.com/nodejs/node/blob/main/AUTHORS
* https://github.com/Homebrew/brew#who-we-are

The spirit is to honor [all contributors](https://github.com/all-contributors/all-contributors), not just the ones who contribute code.

**Comment by MartinThoma** [comment]
If we create that, the following should be given:

* No arguments about order: Either we simply make it alphabetical by last name or something similar
* No claim to be complete: I don't want to forget anybody / go through the commit history

Do we need any rules about edits to that file? How do other projects do it?

**Comment by JianzhengLuo** [comment]
It seemed that GitHub already made a list of contributors at [https://github.com/py-pdf/PyPDF2/graphs/contributors](https://github.com/py-pdf/PyPDF2/graphs/contributors) automatically, even included how much and when they contributed. That could be a source.

**Reply by MartinThoma** [reply]
Ah, right, I forgot about that list - thanks for the reminder!

Would you still appreciate a `CONTRIBUTORS.md` markdown file within the repository?

**Reply by JianzhengLuo** [reply]
Yeah, although I can’t find a reason, why not?😜

Maybe this project will be on other sites, and maybe there won’t a list like the one on GitHub.

**Comment by MartinThoma** [comment]
One question is who we consider to be a contributor.

I suggest the following:

1. Anybody who has an commit in `main` - no matter how big/small or how many. Also if it's via co-authored-by.
2. People who opened helpful issues: (1) Bugs: with complete MCVE (2) Well-described feature requests (3) Potentially some more. Here I would want to have the last call / offer it to people.
3. Community work: This would be exceptional. If I see people being super helpful in answering issues / discussions or being very active on Stackoverflow, I would also consider them being contributors to PyPDF2. Again, here I would want to decide on it.

By far most people would be in (1). They could also open a PR adding themselves (or adding themselves with their first PR).

What do you think?

**Reply by JianzhengLuo** [reply]
Great. Just do it seriously, and properly.

**Comment by MartinThoma** [comment]
A first draft: https://github.com/py-pdf/PyPDF2/pull/1132

@pubpub-zz @MasterOdin @mtd91429 May I add you? Just send me a line / leave a line in the PR with the text I should add :-)

**Reply by pubpub-zz** [reply]
* @pubpub-zz : involved in community development; https://github.com/pubpub-zz
